### PR TITLE
persist eligibility override data on applicant and tax household member

### PR DESCRIPTION
**Pull request by mdkaraman**
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit
)
- [x] Tests for the changes have been added (for bugfixes / features)

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

# What is the ticket # detailing the issue?

Ticket: TT6-185060666

# A brief description of the changes

Current behavior:
Eligibility override data sent to EA from MG is not persisted in the database.  There is a mismatch in field names between the EA model and the latest version of aca_entities.

New behavior:
Eligibility override data sent to EA from MG is persisted in the database.  All field names align between the eligibility override model in EA and the entity/contract in aca_entities.

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable that is used to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name: n/a

- [ ] DC
- [ ] ME

Note:  
This is a system update to how data is modeled and saved to the database.  It is a feature of the system and intended to work with all clients, so does not require a feature flag.  Additionally, updates to aca_entities refs must be included in this PR since there was a change in a field name on an entity and decoupling those changes from this code would break import of the determination results.

### CODE REVIEW

**Review by erubinst — COMMENTED** [review]
New enhancements look good to me.  Looking at original ticket, I see mention of "created_at and updated_at fields are being set to nil" as a bug identified in the original implementation - is this being taken care of with new enhancements?  (Looks like the duplicate determinations issue on the comment was addressed in the MG fix, but also just double checking that comment to make sure all those got addressed).  Thanks!

File: app/models/eligibility_override.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+# Represents an eligibility override for a member
+class EligibilityDetermination
+  include Mongoid::Document
+  include Mongoid::Timestamps
+
+  embedded_in :member_determination
+
+    # The override rule that was applied.
+  field :override_rule, type: Symbol
```
Comment (scaustin34) [inline app/models/eligibility_override.rb:11]: Did we switch this field to be a string?

**Comment by mdkaraman** [comment]
> 

Great catch!  Adding the 'cascade_callbacks: true' option to the MemberDetermination models fixes the issue of nil timestamps.  And yes, tested locally with the latest MG updates and the duplicate record issue no longer present.

**Review by scaustin34 — APPROVED** [review]
Looks good! adds our enhancements to the member_determ object on top of saving the data to EA from in payload

**Review by erubinst — APPROVED** [review]
Looks good to me.

**Review by polographer — APPROVED** [review]

**Pull request merged**